src/extension/background-script/actions/onboard/providerOnboard.ts

```diff
@@ -0,0 +1,22 @@
+import utils from "~/common/lib/utils";
+import { getHostFromSender } from "~/common/utils/helpers";
+import type { MessageAllowanceEnable, Sender } from "~/types";
+
+const providerOnboard = async (
+  message: MessageAllowanceEnable,
```

Comment:
is the message type correct here? `MessageAllowanceEnable`?

Reply:
we are intercepting only enable calls here. this particular interface addresses right message type here i think

```
export interface MessageAllowanceEnable extends MessageDefault {
  origin: OriginData;
  args: {
    host: Allowance["host"];
  };
  action:
    | "public/webln/enable"
    | "public/nostr/enable"
    | "public/liquid/enable"
    | "public/alby/enable";
}
```